### ISSUE TITLE
Tag File - option to Strip path

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9221]
**Describe the bug**
Our doxygen-generated tag files contain internal paths specific to the build machine that are not relevant to the public.
https://root.cern/doc/master/ROOT.tag.gz

**Expected behavior**
It would be nice if an option to strip this path from the output tag file is provided.
(I tried adding /opt/root_src to  STRIP_FROM_PATH but it does not seem to work).

**Screenshots**
```
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<tagfile doxygen_version="1.9.4" doxygen_gitid="70d1c6baf5107dcfaae6fee02ea6b3ef93cea61c*">
  <compound kind="file">
    <name>fillpatterns.C</name>
    <path>/opt/root_src/core/base/doc/macros/</path>
    <filename>fillpatterns_8C.html</filename>
```

**To Reproduce**
[tagpath.zip](https://github.com/doxygen/doxygen/files/8326127/tagpath.zip)
cd into "streamer" and call doxygen there.
Open resulting tag file and the full path to docu will be there, despite STRIP_FROM_PATH being `../`.

**Version**
`1.9.4 (70d1c6baf5107dcfaae6fee02ea6b3ef93cea61c*)`

**Additional context**
https://github.com/root-project/root/issues/10059

**Comment by albert-github** [comment]
I can imagine that it is not desired that an internal path is shown.

I see that the path is set for files and directories (see `dirdef.cpp` and `filedef.cpp`) and that it is read and used in `tagreader.cpp`.
In the tag reader it is used to do a match for include files and to create a filename and a tagid. Most likely usage is to distinguish between files with the same name but in differt directories. I'm not sure whether the  or not the path is used further on.

Maybe a solution would be that in the `dirdef.cpp` and `filedef.cpp` the path is converted to a md5sum.

@doxygen do you see or know whether or not this approach would have have any side effects?

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9226, though with the pull request I also made a small compatibility remark so it is not 100%clear whether this will be the final solution or that we might need to hide the replacement behind a setting.
@ferdymercury maybe you could give the patch a test with your package.

**Comment by ferdymercury** [comment]
Thanks, I run it and I see the paths being converted to md sums. :)
I do not use the Tag file right now externally, thus I cannot check about the potentially broken links.

**Comment by albert-github** [comment]
Pity you cannot test it against an external usage.

Small note: The links are not broken, but just not present anymore.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request https://github.com/doxygen/doxygen/pull/9227 that does not have the problems as shown with #9226. One has to set the `STRIP_FROM_PATH` to the proper path.

**Comment by ferdymercury** [comment]
Thanks, I built it, and it looks great, strip_from_path is respected now :)

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
